src/Servers/Kestrel/Transport.MsQuic/src/Internal/MsQuicConnection.cs

```diff
@@ -0,0 +1,425 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Connections.Abstractions.Features;
+using Microsoft.AspNetCore.Connections.Features;
+using static Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic.Internal.MsQuicNativeMethods;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic.Internal
+{
+    internal class MsQuicConnection : TransportConnection, IQuicStreamListenerFeature, IQuicCreateStreamFeature, IDisposable
+    {
+        private bool _disposed;
+        private readonly MsQuicTransportContext _context;
+        private IntPtr _nativeObjPtr;
+        private ConnectionCallback _callback;
+        private static GCHandle _handle;
+        private readonly IntPtr _unmanagedFnPtrForNativeCallback;
+        private TaskCompletionSource<object> _tcsConnection;
+
+        private readonly Channel<MsQuicStream> _acceptQueue = Channel.CreateUnbounded<MsQuicStream>(new UnboundedChannelOptions
+        {
+            SingleReader = true,
+            SingleWriter = true
+        });
+
+        public MsQuicConnection(MsQuicApi registration, MsQuicTransportContext context, IntPtr nativeObjPtr)
+        {
+            Registration = registration;
+            _context = context;
+            _nativeObjPtr = nativeObjPtr;
+
+            ConnectionCallbackDelegate nativeCallback = NativeCallbackHandler;
+            _unmanagedFnPtrForNativeCallback = Marshal.GetFunctionPointerForDelegate(nativeCallback);
+            SetCallbackHandler(HandleEvent);
+            SetIdleTimeout(TimeSpan.FromMinutes(10));
```

Comment:
I'd prefer you let the transport default stand, unless you have a value from an upper layer to overwrite it.